internal/collector/eth_block_timestamp.go

```diff
@@ -0,0 +1,53 @@
+package collector
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type EthBlockTimestamp struct {
+	rpc  *rpc.Client
+	desc *prometheus.Desc
+}
+
+type blockResult struct {
+	Timestamp hexutil.Uint64
+}
+
+func NewEthBlockTimestamp(rpc *rpc.Client) *EthBlockTimestamp {
+	return &EthBlockTimestamp{
+		rpc: rpc,
+		desc: prometheus.NewDesc(
+			"eth_block_timestamp",
+			"timestamp of the most recent block",
+			nil,
+			nil,
+		),
+	}
+}
+
+func (collector *EthBlockTimestamp) Describe(ch chan<- *prometheus.Desc) {
+	ch <- collector.desc
+}
+
+func (collector *EthBlockTimestamp) Collect(ch chan<- prometheus.Metric) {
+	var bn hexutil.Uint64
+	var result *blockResult
+
+	if err := collector.rpc.Call(&bn, "eth_blockNumber"); err != nil {
+		ch <- prometheus.NewInvalidMetric(collector.desc, err)
+		return
+	}
+
+	if err := collector.rpc.Call(&result, "eth_getBlockByNumber", bn, 0); err != nil {
```

Comment:
I suggest using `latest` as a parameter to that call. You'll get the latest block and do not have to call `eth_blockNumber`. See https://openethereum.github.io/JSONRPC-eth-module#eth_getblockbynumber for details.

Reply:
Great, I missed that one! Updated.